lib/jsdom/living/nodes/HTMLSelectElement-impl.js

```diff
@@ -82,19 +83,69 @@ class HTMLSelectElementImpl extends HTMLElementImpl {
       selected.forEach((option, index) => {
         option._selectedness = index === selected.length - 1;
       });
+      this._selectedOption = selected[selected.length - 1];
     }
   }
 
+  _updateSelected(child) {
+    if (this.hasAttributeNS(null, "multiple")) {
+      return;
+    }
+    if (child.hasAttributeNS(null, "disabled")) {
+      return;
+    }
+    const optArr = this._toOptionArray(child);
+    if (!optArr.length) {
+      return;
+    }
+    const selectedOptions = optArr.filter(opt => opt._selectedness);
+    // If any selected is last selected
+    if (selectedOptions.length) {
+      // pushed at least one element with selected=true, reset
+      this._askedForAReset();
+      return;
+    }
+
+    // No selection so first option is selected
+    if ((!this._selectedOption || !this._selectedOption._selectedness) && this._displaySize === 1) {
+      this._selectedOption = optArr[0];
+      this._selectedOption._selectedness = true;
+    }
+  }
+
+  _toOptionArray(child) {
```

Comment:
Let's factor this out into a standalone function that lives outside the class, instead of a member function, since it doesn't have any dependence on `this`.

Reply:
Done